cluster-agent/controllers/managed-gitops/operation_controller.go

```diff
@@ -60,3 +70,70 @@ func (r *OperationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&managedgitopsv1alpha1.Operation{}).
 		Complete(r)
 }
+
+type garbageCollector struct {
+	db        db.DatabaseQueries
+	k8sClient client.Client
+}
+
+// NewGarbageCollector creates a new instance of garbageCollector for Operations
+func NewGarbageCollector(dbQueries db.DatabaseQueries, client client.Client) *garbageCollector {
+	return &garbageCollector{
+		db:        dbQueries,
+		k8sClient: client,
+	}
+}
+
+// StartGarbageCollector starts a goroutine that removes the expired operations after a specified interval
+func (g *garbageCollector) StartGarbageCollector() {
+	g.startGarbageCollectionCycle()
+}
+
+func (g *garbageCollector) startGarbageCollectionCycle() {
+	go func() {
+		for {
+			// garbage collect the operations after a specified interval
+			<-time.After(garbageCollectionInterval)
+
+			ctx := context.Background()
+			log := log.FromContext(ctx)
+
+			// get failed/completed operations with non-zero gc interval
+			operations := []db.Operation{}
+			err := g.db.ListOperationsToBeGarbageCollected(ctx, &operations)
+			if err != nil {
+				log.Error(err, "failed to list operations ready for garbage collection")
+			}
+
+			g.garbageCollectOperations(ctx, operations, log)
+		}
+	}()
+}
+
+func (g *garbageCollector) garbageCollectOperations(ctx context.Context, operations []db.Operation, log logr.Logger) {
+	for _, operation := range operations {
+		// last_state_update + gc_expiration_time < time.Now
+		if operation.Last_state_update.Add(operation.GetGCExpirationTime()).Before(time.Now()) {
+			// remove the Operation from the DB
+			_, err := g.db.DeleteOperationById(ctx, operation.Operation_id)
+			if err != nil {
+				log.Error(err, "failed to delete operation from DB", "operation_id", operation.Operation_id)
+				continue
+			}
+
+			// remove the Operation resource from the cluster
+			operationCR := &managedgitopsv1alpha1.Operation{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("operation-%s", operation.Operation_id),
+					Namespace: util.GetGitOpsEngineSingleInstanceNamespace(),
+				},
+			}
+			err = g.k8sClient.Delete(ctx, operationCR)
+			if err != nil {
+				log.Error(err, "failed to delete operation from the cluster", "operation_id", operation.Operation_id)
+			}
+
+			log.Info("successfully garbage collected operation", "operation_id", operation.Operation_id)
```

Comment:
If the operation CR fails to be deleted from the cluster, you will end up with two logs. One log msg saying there is an error and another one saying that the operation has been garbage collected. To me, a garbage collected operation means that's deleted from both places (db + cluster).

```suggestion
			err = g.k8sClient.Delete(ctx, operationCR)
			if err != nil {
				log.Error(err, "failed to delete operation from the cluster", "operation_id", operation.Operation_id)
			} else {
				log.Info("successfully garbage collected operation", "operation_id", operation.Operation_id)
			}
```

What happens to the Operation that is failed to be deleted from the cluster but it has successfully been deleted from the database?